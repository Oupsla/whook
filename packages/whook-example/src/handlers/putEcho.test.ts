import initPutEcho, { definition } from './putEcho';
import YError from 'yerror';
import { OpenAPIV3 } from 'openapi-types';

describe('putEcho', () => {
  const log = jest.fn();

  it('should work', async () => {
    const putEcho = await initPutEcho({
      log,
    });
    const response = await putEcho({
      body: (definition.operation.requestBody as OpenAPIV3.RequestBodyObject)
        .content['application/json'].example,
    });

    expect({
      response,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });

  it('should fail when crossing the red line ;)', async () => {
    const putEcho = await initPutEcho({
      log,
    });

    try {
      await putEcho({
        body: 'Big up to Lord Voldemort!',
      });
      throw new YError('E_UNEXPECTED_SUCCESS');
    } catch (err) {
      expect({
        errorCode: err.code,
        errorParams: err.params,
        logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
      }).toMatchSnapshot();
    }
  });
});
