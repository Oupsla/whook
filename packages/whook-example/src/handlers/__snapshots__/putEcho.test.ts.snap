// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`putEcho should fail when crossing the red line ;) 1`] = `
Object {
  "errorCode": undefined,
  "errorParams": undefined,
  "logCalls": Array [
    Array [
      "warning",
      "ðŸ“¢ - Echoing \\"Repeat this!\\"",
    ],
  ],
}
`;

exports[`putEcho should work 1`] = `
Object {
  "logCalls": Array [
    Array [
      "warning",
      "ðŸ“¢ - Echoing \\"Repeat this!\\"",
    ],
  ],
  "response": Object {
    "body": Object {
      "echo": "Repeat this!",
    },
    "status": 200,
  },
}
`;
